### DISCUSSION TITLE
Users that Search by Email not Include realmRoles

### DISCUSSION THREAD

**Discussion by hantsy** [keycloak/keycloak#30782] [Q&A]
I connected to admin API by administrator username/password with client `admin-cli`. 

When using the following codes to search a user by email.

```kotlin
realmResource.users().searchByEmail(email, true).firstOrNull()?.let {
    log.debug("found users by email: ${it.realmRoles}")
    // ...
}
```

The `realmRoles` always return `null`.

But when I use get users by id like the following.

```kotlin 
val userResource = realmResource.users().get(id)
val roles = userResource.roles().realmLevel().listAll().map { it.name }
val user = userResource.toRepresentation()
```

The roles is filled as expected here.

**Comment by Crafoord** [comment]
Hi, I'm getting users by the `GET /admin/realms/{realm}/users` endpoint and also missing realmRoles/clientRoles. Maybe related? It should return a [UserRepresentation[]](https://www.keycloak.org/docs-api/25.0.2/rest-api/#UserRepresentation). But everything in that definition is "optional" so it's very hard to understand what the endpoint can actually return...

**Comment by danielFesenmeyer** [comment]
See my answer here: https://github.com/keycloak/keycloak/discussions/10168#discussioncomment-10528807
Probably it's not really intuitive, but it's how it works currently.
